consensus/ibft/ibft.go

```diff
@@ -1026,6 +1026,13 @@ func (i *Ibft) insertBlock(block *types.Block) error {
 	// broadcast the new block
 	i.syncer.Broadcast(block)
 
+	// On every epoch start create checkpoint
+	if i.IsLastOfEpoch(block.Number()) {
+		if hookErr := i.runHook(CreateCheckpointHook, block.Number(), i.epochSize); hookErr != nil {
+			i.logger.Error(fmt.Sprintf("Unable to run hook %s, %v", CreateCheckpointHook, hookErr))
+		}
```

Comment:
Ideally IBFT must not know about Checkpoint. Now the base is depending on extensions.

Why not use AcceptStateLogHook or InsertBlockHook?

Reply:
Fixed ac388779230090c3ee8dc32f1fa312ecf14305e3